sire/sire.py

```diff
@@ -147,6 +148,22 @@ def _locate_templates():
     raise ValueError(f"No templates found in: {dirs}")
 
 
+def _obtain_github_username():
```

Comment:
this is clever, i like it. ideally this function replaces the `getuser()` call during the interactive prompt too.